src/Functions/array/randomSampleFromArray.cpp

```diff
@@ -0,0 +1,126 @@
+#include <random>
+#include <Columns/ColumnArray.h>
+#include <DataTypes/DataTypeArray.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunction.h>
+#include <Poco/Logger.h>
+#include "Columns/ColumnsNumber.h"
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+/// randomSampleFromArray(arr, k) - Returns k random elements from the input array
+class FunctionRandomSampleFromArray : public IFunction
+{
+public:
+    static constexpr auto name = "randomSampleFromArray";
+
+    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionRandomSampleFromArray>(); }
+
+    String getName() const override { return name; }
+
+    size_t getNumberOfArguments() const override { return 2; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        const DataTypeArray * array_type = checkAndGetDataType<DataTypeArray>(arguments[0].get());
```

Comment:
You might want to look into `FunctionArgumentDescriptors` and `validateFunctionArgumentTypes()` to validate the input types. They are more high-level than the current code.

Reply:
Thanks, I will take a look.

Reply:
I tried today once, but the function is not recognized. 
I will try it again now.

Reply:
done